.github/workflows/tests.yml

```
@@ -314,7 +314,7 @@ jobs:
 
       - name: Check for flaky tests report
         id: check-flaky-spec-report
-        if: github.repository == 'discourse/discourse' && env.DISCOURSE_TURBO_RSPEC_RETRY_AND_LOG_FLAKY_TESTS == '1'
+        if: github.repository == 'discourse/discourse' && github.ref == 'refs/heads/main' && env.DISCOURSE_TURBO_RSPEC_RETRY_AND_LOG_FLAKY_TESTS == '1'
```

I tend to prefer `github.ref_name` for readability, but don't have strong feelings either way. 🙂

```suggestion
        if: github.repository == 'discourse/discourse' && github.ref_name == 'main' && env.DISCOURSE_TURBO_RSPEC_RETRY_AND_LOG_FLAKY_TESTS == '1'
```